src/df/df.rs

```diff
@@ -0,0 +1,103 @@
+/*
+ Copyright (c) 2023 Uber Technologies, Inc.
+
+ <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ except in compliance with the License. You may obtain a copy of the License at
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+use getset::Getters;
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// Trait for states (sigma), which store the states of the analysis at each program point
+pub trait Sigma: Clone {
```

Comment:
maybe this trait could extend `Eq` and `PartialEq` ? then u will get the default equals for free, and will not have to create and override `is_equal`


To implement the `Eq` you can - https://doc.rust-lang.org/std/cmp/trait.Eq.html at the appropriate struct